# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/GfxInfoBase.cpp
# Commit: 40fa8e4b06c8
# Full Hash: 40fa8e4b06c85e9e241da6e56b62a0b2bde7427c
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-03-28 17:08:58
# Regressor Bug: 1887729
# File Overlap Count: 2
# Description:
#   Bug 1887729 - Implement context lost/restored support for CanvasRenderingContext2D. r=webidl,gfx-reviewers,smaug,lsalzman
#   
#   Remote canvas can run in the GPU process, and if the GPU process
#   crashes, we need to notify the application using canvas. Historically we
#   just failed, and the application may have been able to continue drawing
# ==============================================================================

diff -r 469884ac63bc -r 40fa8e4b06c8 widget/GfxInfoBase.cpp
--- a/widget/GfxInfoBase.cpp	Thu Mar 28 01:16:16 2024 +0000
+++ b/widget/GfxInfoBase.cpp	Thu Mar 28 01:51:23 2024 +0000
@@ -2161,6 +2161,18 @@
   return NS_OK;
 }
 
+NS_IMETHODIMP
+GfxInfoBase::GetUsingRemoteCanvas(bool* aOutValue) {
+  *aOutValue = gfx::gfxVars::RemoteCanvasEnabled();
+  return NS_OK;
+}
+
+NS_IMETHODIMP
+GfxInfoBase::GetUsingAcceleratedCanvas(bool* aOutValue) {
+  *aOutValue = gfx::gfxVars::UseAcceleratedCanvas2D();
+  return NS_OK;
+}
+
 NS_IMETHODIMP_(int32_t)
 GfxInfoBase::GetMaxRefreshRate(bool* aMixed) {
   if (aMixed) {